lib/optimizely/odp/odp_config.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2022, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'optimizely/logger'
+
+module Optimizely
+  class OdpConfig
+    # Contains configuration used for ODP integration.
+    #
+    # @param api_host - The host URL for the ODP audience segments API (optional). If not provided, SDK will use the default host in datafile.
+    # @param api_key - The public API key for the ODP account from which the audience segments will be fetched (optional). If not provided, SDK will use the default publicKey in datafile.
+    # @param segments_to_check - An array of all ODP segments used in the current datafile (associated with api_host/api_key).
+    # @param odp_enabled - A boolean value indicating whether odp is enabled for the project or not.
+    def initialize(api_key = nil, api_host = nil, segments_to_check = [])
+      @api_key = api_key
+      @api_host = api_host
+      @segments_to_check = segments_to_check
+      @odp_enabled = !api_key.nil? && !api_host.nil? ? true : false
+      @mutex = Mutex.new
+    end
+
+    # Replaces the existing configuration
+    #
+    # @param api_host - The host URL for the ODP audience segments API (optional). If not provided, SDK will use the default host in datafile.
+    # @param api_key - The public API key for the ODP account from which the audience segments will be fetched (optional). If not provided, SDK will use the default publicKey in datafile.
+    # @param segments_to_check - An array of all ODP segments used in the current datafile (associated with api_host/api_key).
+    #
+    # @return - True if the provided values were different than the existing values.
+
+    def update(api_key = nil, api_host = nil, segments_to_check = [])
+      @mutex.synchronize do
+        @odp_enabled = !api_key.nil? && !api_host.nil? ? true : false
```

Comment:
Might sound like being nitpicking but just curios how it will play out if someone later passes a null apiKey or host through the setter, should this be updated to a method that evaluates the current values when called?